DOCS-859: Clarify viam-cartographer as modular resource (#1403)
viamrobotics/docs: docs/services/slam/cartographer/_index.md

---
title: "Cartographer Modular Resource"
linkTitle: "Cartographer"
weight: 70
type: "docs"
description: "Configure a SLAM service with the Cartographer modular resource."
tags: ["slam", "services"]
icon: "/services/img/icons/slam.svg"
aliases:
  - "/services/slam/run-slam-cartographer"
# SMEs: Kat, Jeremy
---

[The Cartographer Project](https://github.com/cartographer-project) performs dense SLAM using LIDAR data.

> To easily integrate Cartographer into the Viam ecosystem, use the [`viam-cartographer`](https://github.com/viamrobotics/viam-cartographer) library, which wraps Cartographer as a [modular resource](/program/extend/modular-resources/). `viam-cartographer` provides the `cartographer-module` module, which includes the `viam:slam:cartographer` custom SLAM service [model](/program/extend/modular-resources/#models).

Reviewer: ```suggestion
To easily integrate Cartographer into the Viam ecosystem, use the [`viam-cartographer`](https://github.com/viamrobotics/viam-cartographer) library, which wraps Cartographer as a [modular resource](/program/extend/modular-resources/). 
`viam-cartographer` provides the `cartographer-module` module, which includes the `viam:slam:cartographer` custom SLAM service [model](/program/extend/modular-resources/#models).
```

Author: 🤦 Thank you!

Reviewer: (opt) It might be reasonable to try to combine this into one sentence if you could, these sentences are still a bit awkward sounding but i think it's inevitable with this situation 😆

Author: Totally -- I tried a bunch! I think I'm gonna go with what we've got while I keep learning the intricacies of our modular approach. Will revisit once I've got a better grasp!

---
title: "Cartographer Modular Resource"
linkTitle: "Cartographer"
weight: 70
type: "docs"
description: "Configure a SLAM service with the Cartographer modular resource."
tags: ["slam", "services"]
icon: "/services/img/icons/slam.svg"
aliases:
  - "/services/slam/run-slam-cartographer"
# SMEs: Kat, Jeremy
---

[The Cartographer Project](https://github.com/cartographer-project) performs dense SLAM using LIDAR data.

To easily integrate Cartographer into the Viam ecosystem, Viam provides the [`viam-cartographer`](https://github.com/viamrobotics/viam-cartographer) module which wraps Cartographer as a [modular resource](/program/extend/modular-resources/) in the form of a custom SLAM service.

> This module provides the `cartographer-module` binary, which runs in addition to `viam-server`.

Author: I really need to include mention of `cartographer-module` here, since this rest of this page uses this term over the others introduced in this paragraph.

> This module provides the `cartographer-module` binary, which runs in addition to `viam-server`.

Reviewer: ```suggestion
```
I'm not sure this sentence is helpful if we are not clarifying where the two are running-- seems slightly confusing

Author: Gotcha. I really need to include mention of `cartographer-module` here, since this rest of this page uses this term over the others introduced in this paragraph. Otherwise there's nothing to connect that:

the module `viam-cartographer` installs the binary `cartographer-module`.

Reviewer: Ah ok, great point. I was looking over this again and realized what could actually be confusing is referring to `viam-cartographer` as the module-- `cartographer-module` is the module name. The " `cartographer` library" (viam:slam:cartographer) " is the model of SLAM service provided by this module, hence it itself is the modular resource. If you make that clarification this will be perfect, agree more info is needed!

---
title: "Cartographer Modular Resource"
linkTitle: "Cartographer"
weight: 70
type: "docs"
description: "Configure a SLAM service with the Cartographer modular resource."
tags: ["slam", "services"]
icon: "/services/img/icons/slam.svg"
aliases:
  - "/services/slam/run-slam-cartographer"
# SMEs: Kat, Jeremy
---

services/img/icons/slam.svg

[The Cartographer Project](https://github.com/cartographer-project) performs dense SLAM using LIDAR data.

> To easily integrate Cartographer into the Viam ecosystem, Viam provides [`viam-cartographer`](https://github.com/viamrobotics/viam-cartographer), a [modular resource](/program/extend/modular-resources/) which wraps Cartographer. This resource provides the `cartographer-module` binary for use directly on your SLAM-enabled robot.

Reviewer: ```suggestion
To easily integrate Cartographer into the Viam ecosystem, Viam provides [`viam-cartographer`](https://github.com/viamrobotics/viam-cartographer), a module which provides Cartographer as a
[modular resource](/program/extend/modular-resources/), a new model of SLAM service.
```
Particularly last part of sentnece is optional but I think the change to module is accurate? Basically the slam service itself is the resource 
A module can have multiple resources but this one is just the new service type

Author: Thank you! You're right!

[The Cartographer Project](https://github.com/cartographer-project) performs dense SLAM using LIDAR data.

> To easily integrate Cartographer into the Viam ecosystem, Viam provides [`viam-cartographer`](https://github.com/viamrobotics/viam-cartographer), a [modular resource](/program/extend/modular-resources/) which wraps Cartographer. This resource provides the `cartographer-module` binary for use directly on your SLAM-enabled robot.

Reviewer: ```suggestion
To easily integrate Cartographer into the Viam ecosystem, Viam provides [`viam-cartographer`](https://github.com/viamrobotics/viam-cartographer), a module which provides Cartographer as a
[modular resource](/program/extend/modular-resources/).
```
Suggesting as I think something like this would help users to understand the modular resource part, it's somewhat more in line with our other phrasing around modules / modular resources? the modules part has to be changed but rest is opt
